### PR TITLE
Configure https proxy + ca bundle

### PR DESCRIPTION
**Pull request by Ripolin**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Allow osc ccm to manipulate osc api through an https proxy when k8s cluster is not directly exposed on internet.

**Special notes for your reviewer**:

Update helm assets to add:
* an http_proxy env var
* an no_proxy env var (default value exclude all [private ipv4 adresses](https://en.wikipedia.org/wiki/Private_network))
* a volume to share a custom ca bundle to use

**Does this PR introduce a user-facing change?**:

NONE

### CODE REVIEW

File: deploy/k8s-osc-ccm/values.yaml [hunk 1]
```diff
@@ -25,4 +25,13 @@ podLabels: {}
...
+tolerations: []
+
+# -- HTTP proxy configuration example 
+# caBundle:
+#   name: my-ca-bundle
+#   key: my-ca-bundle.pem
+
+# httpsProxy: https://my-proxy.foo.bar
+
+noProxy: 10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,169.254.169.254
```
Comment (outscale-vbr) [inline deploy/k8s-osc-ccm/values.yaml:37]: Can you also comment it ?

Comment (Ripolin) [reply deploy/k8s-osc-ccm/values.yaml]: Done in commit 3032aee6a3428b064b6ef1cca4c1323123bcafa2

File: deploy/k8s-osc-ccm/values.yaml [hunk 2]
```diff
@@ -25,4 +25,11 @@ podLabels: {}
 # -- Assign Pod to Nodes (see [kubernetes doc](https://kubernetes.io/docs/tasks/configure-pod-container/assign-pods-nodes/))
 nodeSelector: {}
 # -- Pod tolerations (see [kubernetes doc](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/))
-tolerations: []
\ No newline at end of file
+tolerations: []
+
+# -- HTTP proxy configuration example
```
Comment (outscale-vbr) [inline deploy/k8s-osc-ccm/values.yaml:30]: We use helm docs.
In order to have a clear and well written docs, Can you set please values and comments in this form:
```
# -- HTTP proxy configuration example 
caBundle:
# -- Explain caBundle Name value
  name: ''
# -- Explain caBundle Key value
  key: ''
# -- Explain httpsProxy value
httpsProxy: ''
# -- Explain noProxy value
noProxy: ''
```

**Comment by outscale-vbr** [comment]
@Ripolin 
There are also merge conflict.
Please also update chart version (if needed) in [Chart.yaml](https://github.com/outscale/cloud-provider-osc/blob/OSC-MIGRATION/deploy/k8s-osc-ccm/Chart.yaml)
And generate helm doc make helm-docs.

**Comment by Ripolin** [comment]
Conflicts resolved and values.yaml updated

**Review by outscale-vbr — APPROVED** [review]

**Pull request merged**